src/scenes/TradeHub/index.js

```diff
@@ -82,141 +73,142 @@ class TradeHub extends Component {
+      return;
+    }
+
+    if (this.shouldSell(price)) this.sell(price);
+  };
+
+  setBoughtPrice = price => this.setState({ boughtPrice: price });
+
+  setDiffPercentage = percentage =>  this.setState({ diffPercentage: percentage / 100 });
+
+  setSellEnabled = value => this.setState({ sellEnabled: value });
+
+  isHighestPrice = price => price > this.state.highestPrice;
+
+  setHighestPrice = price => this.setState({ highestPrice: price });
+
+  shouldSell = (price) => {
+    const { sellEnabled, hasSold, highestPrice, diffPercentage } = this.state;
+    if (sellEnabled && !hasSold) {
+      return price <= highestPrice - highestPrice * diffPercentage;
+    }
+  };
+
+  sell = (price) => {
+    alert('SOLD at: ' + price);
+    this.setState({
+      hasSold: true
+    });
+  };
 
-  render() {
-    const selectedCrypto = this.state.selectedCrypto;
+  render = () => {
+    const { sellEnabled, selectedCrypto, cryptoList, boughtPrice } = this.state;
+    const diffPercentage = this.state.diffPercentage * 100;
     const currentPrice = this.state.currentPrice.toFixed(6);
     const highestPrice = this.state.highestPrice.toFixed(6);
     const sellPrice = (highestPrice - highestPrice * this.state.diffPercentage).toFixed(6);
-    const cryptoList = this.state.cryptoList;
+
+    const sellOptions = [
+      {
+        label: 'Enable',
+        value: true,
+        color: 'success'
+      },
+      {
+        label: 'Disable',
+        value: false,
+        color: 'danger'
+      }
+    ];
 
     return (
       <div>
         <Header />
 
-        <div className="container">
+        <Container>
           <HorizontalTabList list={cryptoList} selectedItem={this.state.selectedCrypto} changeSelected={this.changeSelectedCrypto}/>
-          <div className="row">
-            <div className="col-sm"><InfoPanel title={selectedCrypto.quoteAsset + " " + currentPrice} description={selectedCrypto.baseAsset + " current price"}/></div>
-            <div className="col-sm"><InfoPanel title={selectedCrypto.quoteAsset + " "  + highestPrice} description={selectedCrypto.baseAsset + " hightest price since bought"}/></div>
-            <div className="col-sm"><InfoPanel title={selectedCrypto.quoteAsset + " "  + sellPrice} description={selectedCrypto.baseAsset + " price to sell on"}/></div>
-          </div>
-        </div>
-
-        <div className="container mt-3">
-          <div className="row">
-            <div className="col-sm"><Chart selectedCrypto={this.state.selectedCrypto} /></div>
-          </div>
-        </div>
+          <Row>
+            <Col>
+              <InputPanel
+                name="bought_price"
+                value={boughtPrice}
+                step="0.01"
+                onChange={this.setBoughtPrice}
+                title="Bought"
+                description={"Price in " + selectedCrypto.quoteAsset + " at which you bought " + selectedCrypto.baseAsset}
+                placeholder="Bought price"/>
```

Comment:
Why the `/>` on the same line?
It's actually [preferred](https://github.com/airbnb/javascript/tree/master/react#alignment) when you put it on the next line.

So like this:
```javascript
<InputPanel
  name="bought_price"
  value={boughtPrice}
  {/* ... */}
/>
```

Reply:
Agree, following indenting rules as specified here: https://github.com/airbnb/javascript/tree/master/react